src/main/java/de/rub/nds/modifiablevariable/Logger/CustomLayout.java

```diff
@@ -0,0 +1,562 @@
+
+        public CustomLayout.SerializerBuilder setPatternSelector(PatternSelector patternSelector) {
+            this.patternSelector = patternSelector;
+            return this;
+        }
+
+        public CustomLayout.SerializerBuilder setAlwaysWriteExceptions(
+                boolean alwaysWriteExceptions) {
+            this.alwaysWriteExceptions = alwaysWriteExceptions;
+            return this;
+        }
+
+        public CustomLayout.SerializerBuilder setDisableAnsi(boolean disableAnsi) {
+            this.disableAnsi = disableAnsi;
+            return this;
+        }
+
+        public CustomLayout.SerializerBuilder setNoConsoleNoAnsi(boolean noConsoleNoAnsi) {
+            this.noConsoleNoAnsi = noConsoleNoAnsi;
+            return this;
+        }
+    }
+
+    private static class NewCustSerializer
+            implements AbstractStringLayout.Serializer,
+                    AbstractStringLayout.Serializer2,
+                    LocationAware {
+        private final PatternFormatter[] formatters;
+        private final RegexReplacement replace;
+
+        private NewCustSerializer(PatternFormatter[] formatters, RegexReplacement replace) {
+            this.formatters = formatters;
+            this.replace = replace;
+        }
+
+        public String toSerializable(LogEvent event) {
+            StringBuilder sb = AbstractStringLayout.getStringBuilder();
+
+            String var3;
+            try {
+                var3 = this.toSerializable(event, sb).toString();
+            } finally {
+                AbstractStringLayout.trimToMaxSize(sb);
+            }
+
+            return var3;
+        }
+
+        public StringBuilder toSerializable(LogEvent event, StringBuilder buffer) {
+            int len = this.formatters.length;
+
+            for (int i = 0; i < len; ++i) {
+                this.formatters[i].format(event, buffer);
+            }
+
+            if (this.replace != null) {
+                String str = buffer.toString();
+                str = this.replace.format(str);
+                buffer.setLength(0);
+                buffer.append(str);
+            }
+
+            /** Added section to parse ByteArrays to the correct output format. */
+            Class<?> bArrayClass = (new byte[1]).getClass();
+
+            // Iterate over each parameter of a {@Link LogEvent} to find all ByteArrays
+            for (Object param : event.getMessage().getParameters()) {
+
+                // Replace all ByteArrays with the String representation of the ByteArray calculated
+                // by the ArrayConverter.
+                if (bArrayClass == param.getClass()) {
+                    buffer.replace(
+                            buffer.indexOf(Arrays.toString((byte[]) param)),
+                            buffer.indexOf(Arrays.toString((byte[]) param))
+                                    + Arrays.toString((byte[]) param).length(),
+                            ArrayConverter.bytesToHexString((byte[]) param));
```

Comment:
IIRC this should be configurable with pretty printing and the initial new line via pattern attributes.

Reply:
Now it is possible to set the pretty printing and the initinal new line in the logging configuration. By default is that they are not used.